app/controllers/users.rb

```diff
@@ -0,0 +1,21 @@
+get "/users/new" do
+  p params[:user]
+  @user = User.new()
+  erb :"users/new"
+end
+
+post "/users/new" do
+  info = params[:user]
+  if info["password"] == info["password_confirmation"]
+    info.delete("password_confirmation")
+    @user = User.new(info)
+  else
+    redirect ("/users/new")
```

Comment:
When there are errors dont redirect, render right away so you can use your model validation errors without having to hack them into the next request.